File: src/main/java/com/wikia/tibia/repositories/BookRepository.java
package com.wikia.tibia.repositories;

import com.wikia.tibia.gateways.BookGateway;
import com.wikia.tibia.objects.Book;

public class BookRepository extends WikiObjectRepository {

    public BookRepository() {
        super(Book.class, new BookGateway());
    }
}

File: src/main/kotlin/com/wikia/tibia/repositories/BookRepository.kt
package com.wikia.tibia.repositories

import com.wikia.tibia.enums.Contract
import com.wikia.tibia.gateways.WikiObjectGateway
import com.wikia.tibia.objects.Book

class BookRepository : WikiObjectRepository<Book>(
    wikiObjectClass = Book::class.java,
    wikiObjectGateway = WikiObjectGateway(Contract.BOOKS)
)
